### PR TITLE
Separate out `application.conf` into smaller `.conf` files

### PR DESCRIPTION
**Pull request by MichaelZetune**
### Description

Refactors the very long `application.conf` file into smaller `.conf` files in a new `helper/` directory.

This will help modularize the config and make it more readable.

### CODE REVIEW

File: server/conf/application.conf [hunk 1]
```diff
@@ -15,174 +15,26 @@
...
+include "helper/modules.conf"
+include "helper/secret-key.conf"
+include "helper/security.conf"
+include "helper/webservice.conf"
+include "helper/whitelabel.conf"
+
 ## filters/LoggingFilter configuration. Overrided in application.dev.conf.
 filters.LoggingFilter.enable_request_session_logging = false
 
-## Akka
```
Comment (MichaelZetune) [inline server/conf/application.conf:21]: Now in `akka.conf`.

File: server/conf/application.conf [hunk 2]
```diff
@@ -15,174 +15,26 @@
...
-  civic_entity_full_name = "City of TestCity"
-  civic_entity_full_name = ${?WHITELABEL_CIVIC_ENTITY_FULL_NAME}
-
-  # The URL of a 32x32 or 16x16 pixel [favicon](https://developer.mozilla.org/en-US/docs/Glossary/Favicon)
-  # image, in GIF, PNG, or ICO format.
-  favicon_url = "https://civiform.us/favicon.png"
-  favicon_url = ${?FAVICON_URL}
-}
-
-## Secret key
```
Comment (MichaelZetune) [inline server/conf/application.conf:65]: Now in `secret-key.conf`.

File: server/conf/application.conf [hunk 3]
```diff
@@ -15,174 +15,26 @@
...
-# http://www.playframework.com/documentation/latest/ApplicationSecret
-# ~~~~~
-# The secret key is used to sign Play's session cookie.
-# This must be changed for production, but we don't recommend you change it in this file.
-play.http.secret.key = "changeme"
-play.http.secret.key = ${?SECRET_KEY}
-
-auth {
```
Comment (MichaelZetune) [inline server/conf/application.conf:73]: Now in `auth.conf`.

File: server/conf/application.conf [hunk 4]
```diff
@@ -210,192 +63,6 @@ staging_ti_notification_mailing_list = ${?STAGING_TI_LIST}
...
-  enabled += modules.MainModule
-  enabled += modules.DatabaseSeedModule
-  enabled += modules.ProgramCreationModule
-  enabled += modules.DurableJobModule
-
-  # If there are any built-in modules that you want to disable, you can list them here.
-  #disabled += ""
-}
-
-## IDE
```
Comment (MichaelZetune) [inline server/conf/application.conf:240]: Now in `ide.conf`.

File: server/conf/application.conf [hunk 5]
```diff
@@ -210,192 +63,6 @@ staging_ti_notification_mailing_list = ${?STAGING_TI_LIST}
...
-# https://www.playframework.com/documentation/latest/IDE
-# ~~~~~
-# Depending on your IDE, you can add a hyperlink for errors that will jump you
-# directly to the code location in the IDE in dev mode. The following line makes
-# use of the IntelliJ IDEA REST interface:
-#play.editor="http://localhost:63342/api/file/?file=%s&line=%s"
-
-## Internationalisation
```
Comment (MichaelZetune) [inline server/conf/application.conf:248]: Now in `internationalization.conf`.

File: server/conf/application.conf [hunk 6]
```diff
@@ -210,192 +63,6 @@ staging_ti_notification_mailing_list = ${?STAGING_TI_LIST}
...
-  langs = ${?CIVIFORM_SUPPORTED_LANGUAGES}
-
-  # Whether the language cookie should be secure or not
-  langCookieSecure = true
-
-  # Whether the HTTP only attribute of the cookie should be set to true
-  langCookieHttpOnly = true
-}
-
-## Play HTTP settings
```
Comment (MichaelZetune) [inline server/conf/application.conf:278]: Now in `http.conf`.

File: server/conf/application.conf [hunk 7]
```diff
@@ -210,192 +63,6 @@ staging_ti_notification_mailing_list = ${?STAGING_TI_LIST}
...
-  flash {
-    # Sets the cookie to be sent only over HTTPS.
-    secure = true
-
-    # Sets the cookie to be accessed only by the server.
-    httpOnly = true
-  }
-}
-
-## WS (HTTP Client)
```
Comment (MichaelZetune) [inline server/conf/application.conf:338]: Now in `webservice.conf`.

File: server/conf/application.conf [hunk 8]
```diff
@@ -210,192 +63,6 @@ staging_ti_notification_mailing_list = ${?STAGING_TI_LIST}
...
-    #debug.handshake = true
-    #trustManager = {
-    #  stores = [
-    #    { type = "PKCS12", path = "exampletrust.p12" }
-    #  ]
-    #}
-  }
-}
-
-## Cache
```
Comment (MichaelZetune) [inline server/conf/application.conf:370]: Now in `cache.conf`.

File: server/conf/application.conf [hunk 9]
```diff
@@ -404,238 +71,15 @@ server_metrics.enabled = ${?CIVIFORM_SERVER_METRICS_ENABLED}
 # The Prometheus Hotspot library provides some default collectors for garbage collection, memory pool, etc
 play-prometheus-filters.register-default-hotspot-collectors = true
 
-## Filter Configuration
```
Comment (MichaelZetune) [inline server/conf/application.conf:407]: Now in `filters.conf`.

File: server/conf/application.conf [hunk 10]
```diff
@@ -404,238 +71,15 @@ server_metrics.enabled = ${?CIVIFORM_SERVER_METRICS_ENABLED}
...
-  # Play provides a filter that lets you configure which hosts can access your application.
-  # This is useful to prevent cache poisoning attacks.
-  hosts {
-    # Allow requests to localhost on ports 9000 (dev) and 19001 (default test) ports
-    # Also allow any host until we get our own domain.
-    allowed = ["localhost:9000", "localhost:9999", "localhost:19001", "."]
-  }
-}
-
-## Evolutions
```
Comment (MichaelZetune) [inline server/conf/application.conf:510]: Now in `database.conf`.

File: server/conf/application.conf [hunk 11]
```diff
@@ -404,238 +71,15 @@ server_metrics.enabled = ${?CIVIFORM_SERVER_METRICS_ENABLED}
...
-db.default.url = ${?DB_JDBC_STRING}
-db.default.username = ${?DB_USERNAME}
-db.default.password = ${?DB_PASSWORD}
-
 play.assets {
   path = "/public"
   urlPrefix = "/assets"
 }
 
-## EBean
```
Comment (MichaelZetune) [inline server/conf/application.conf:602]: Now in `database.conf`.

File: server/conf/application.conf [hunk 12]
```diff
@@ -404,238 +71,15 @@ server_metrics.enabled = ${?CIVIFORM_SERVER_METRICS_ENABLED}
...
-# You can declare as many Ebean servers as you want.
-# By convention, the default server is named `default`
-ebean.default = "models.*"
-
-## AWS
```
Comment (MichaelZetune) [inline server/conf/application.conf:607]: Now in `cloud.conf`.

File: server/conf/application.conf [hunk 13]
```diff
@@ -657,85 +101,3 @@ api_applications_list_max_page_size = ${?CIVIFORM_API_APPLICATIONS_LIST_MAX_PAGE
 
 # deterministic reporting stats for browser tests
 reporting_use_deterministic_stats = false
-
-## Durable Jobs
```
Comment (MichaelZetune) [inline server/conf/application.conf:661]: Now in `jobs.conf`.

File: server/conf/application.conf [hunk 14]
```diff
@@ -657,85 +101,3 @@ api_applications_list_max_page_size = ${?CIVIFORM_API_APPLICATIONS_LIST_MAX_PAGE
...
-admin_reporting_ui_enabled = ${?CIVIFORM_ADMIN_REPORTING_UI_ENABLED}
-
-# If enabled allows for setting of feature flags via HTTP.
-# This is explicitly not recommended for production and intended to aid
-# development and pre-release evaluation of features. In fact it is unlikely to
-# work as expected in production if there are multiple servers.
-feature_flag_overrides_enabled = false
-feature_flag_overrides_enabled = ${?FEATURE_FLAG_OVERRIDES_ENABLED}
-
-# esri address services
```
Comment (MichaelZetune) [inline server/conf/application.conf:722]: Now in `esri.conf`.

File: server/conf/application.conf [hunk 15]
```diff
@@ -657,85 +101,3 @@ api_applications_list_max_page_size = ${?CIVIFORM_API_APPLICATIONS_LIST_MAX_PAGE
...
-esri_address_service_area_validation_enabled = false
-esri_address_service_area_validation_enabled = ${?ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ENABLED}
-esri_address_service_area_validation_labels = ${?ESRI_ADDRESS_SERVICE_AREA_VALIDATION_LABELS}
-esri_address_service_area_validation_ids = ${?ESRI_ADDRESS_SERVICE_AREA_VALIDATION_IDS}
-esri_address_service_area_validation_urls = ${?ESRI_ADDRESS_SERVICE_AREA_VALIDATION_URLS}
-esri_address_service_area_validation_attributes =  ${?ESRI_ADDRESS_SERVICE_AREA_VALIDATION_ATTRIBUTES}
-# the number of tries to make if Esri API returns an error - defaults to 3
-esri_external_call_tries = ${?ESRI_EXTERNAL_CALL_TRIES}
-
-# Common Intake Form flags
```
Comment (MichaelZetune) [inline server/conf/application.conf:737]: Now in `feature-flags.conf`.

File: server/conf/application.conf [hunk 16]
```diff
@@ -404,238 +71,15 @@ server_metrics.enabled = ${?CIVIFORM_SERVER_METRICS_ENABLED}
...
-azure.blob.container=civiform-local-blob
-azure.blob.container=${?AZURE_STORAGE_ACCOUNT_CONTAINER}
-azure.local.connection="DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://azurite:10000/devstoreaccount1;"
-azure.local.connection=${?AZURE_LOCAL_CONNECTION_STRING}
-
 ## Time zones
 civiform.time.zoneid="America/Los_Angeles"
 civiform.time.zoneid=${?CIVIFORM_TIME_ZONE_ID}
 
-## Set cloud services
```
Comment (MichaelZetune) [inline server/conf/application.conf:631]: Now in `cloud.conf`.

**Comment by MichaelZetune** [comment]
@avrittrohwer, I'm not sure how exactly to validate these changes aside from seeing that `bin/run-dev` still works. I don't see a validation mechanism in Play's docs:

https://www.playframework.com/documentation/2.0/Configuration

Do you know of a way to print out the final contents of `application.conf` so we can diff them before/after?

**Review by avrittrohwer — CHANGES_REQUESTED** [review]
Overall I think this is a good direction, I think having slightly less helper files may be a better balance for readability tho.

What if we consolidate akka, cache, filters, http, ide, modules, secret-key and webservice into a framework.conf?

**Review by bion — CHANGES_REQUESTED** [review]
I think splitting up some of the conf file could be helpful, but one thing I don't want to lose is the main application.conf file containing the things people will expect to find in it when reading the Play Framework docs, which suggests leaving framework-related conf items (i.e. everything nested under play, akka, database, db, filters, modules) in the main file. Otherwise I anticipate regular questions when people look for something they expect to find in that file and don't immediately see it.

**Comment by MichaelZetune** [comment]
I've cleaned this up so that Play framework related vars remain in the main application.conf file.

I think the current state of this PR balances readability and modularity well now. Feel free to let me know if this PR doesn't make sense overall, though - happy to revert.

**Review by avrittrohwer — APPROVED** [review]
nit: put the ide.conf settings in application.conf

**Comment by MichaelZetune** [comment]
> nit: put the ide.conf settings in application.conf

Done.

**Review by bion — APPROVED** [review]

**Comment by MichaelZetune** [comment]
I checked out `main` and this branch and printed out the vars as suggested by @avrittrohwer, then piped them to a txt file. The diffs are here:

https://www.diffchecker.com/pZ0qLvFv/

Everything appears to check out on the vars themselves.

**Pull request merged**